### ISSUE TITLE
 loading Epics CA DLL failed on Windows

### ISSUE THREAD

**Issue by Jo-Leh** [pyepics/pyepics#264]
There seems to be an issue with the newest version on Windows.
I get the following message on a github runner version: '2.317.0', using 'windows-latest'

I did not get this error in version 3.5.2, only when using 3.5.3

```
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\epics\ca.py", line 375, in initialize_libca
    load_dll(find_libCom())
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\ctypes\__init__.py", line 454, in LoadLibrary
    return self._dlltype(name)
           ^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\ctypes\__init__.py", line 366, in __init__
    if '/' in name or '\\' in name:
       ^^^^^^^^^^^
TypeError: argument of type 'WindowsPath' is not iterable
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\epics\ca.py", line 375, in initialize_libca
    load_dll(find_libCom())
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\ctypes\__init__.py", line 454, in LoadLibrary
    return self._dlltype(name)
           ^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\ctypes\__init__.py", line 366, in __init__
    if '/' in name or '\\' in name:
       ^^^^^^^^^^^
TypeError: argument of type 'WindowsPath' is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

# somewhere in my code
    import ophyd
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\ophyd\__init__.py", line 89, in <module>
    set_cl()
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\ophyd\__init__.py", line 24, in set_cl
    set_cl(c_type, pv_telemetry=pv_telemetry)
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\ophyd\__init__.py", line 51, in set_cl
    shim.setup(logger)
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\ophyd\_pyepics_shim.py", line 203, in setup
    thread_class=PyepicsCallbackThread, context=ca.current_context(), logger=logger
                                                ^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\epics\ca.py", line 585, in wrapper
    initialize_libca()
  File "C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\epics\ca.py", line 378, in initialize_libca
    raise ChannelAccessException('loading Epics CA DLL failed: ' + str(exc))
epics.ca.ChannelAccessException: loading Epics CA DLL failed: argument of type 'WindowsPath' is not iterable
```

**Comment by newville** [comment]
@Jo-Leh Thanks for the report, and yikes!  

I think I understand the problem.   The switch to using `importlib.resources` for finding package files did mean that Pathlib paths are used now.  But `ctypes LoadLibrary` wants a Posix Path string, not a Python Pathlib object (but this appears to be a problem only on Windows).    

Normally, I test by hand on Windows, but with APS still in upgrade mode and my beamline slightly torn apart, I did not test on Windows this time.  

I'll want to do a few more tests but hope to push a fix later today.

**Comment by newville** [comment]
... as it turns out, this appears to be fixed with Python 3.12.   But, I pushed a fix to GitHub master.  Tests pass for me on Windows with both Python 3.11 and 3.12.   GitHub actions reports success, so I will tag 3.5.4 and push to PyPI shortly.
